syne_tune/optimizer/schedulers/searchers/regularized_evolution.py

```diff
@@ -0,0 +1,138 @@
+        https://colab.research.google.com/github/google-research/google-research/blob/master/evolution/regularized_evolution_algorithm/regularized_evolution.ipynb
+
+
+        Parameters
+        ----------
+        configspace: dict
+            Configuration space for trial evaluation function
+        metric : str
+            Name of metric to optimize, key in result's obtained via
+            `on_trial_result`
+        mode : str
+            Mode to use for the metric given, can be 'min' or 'max', default to 'min'.
+        population_size : int
+            Size of the population.
+        sample_size : int
+            Size of the candidate set to obtain a parent for the mutation.
+        random_seed : int
+            Seed for the random number generation. If set to None, use random seed.
+        """
+
+        super(RegularizedEvolution, self).__init__(configspace, metric, points_to_evaluate=points_to_evaluate,
+                                                   random_seed=random_seed)
+        self.mode = mode
+        self.population_size = population_size
+        self.sample_size = sample_size
+        self.population = deque()
+
+    def mutate_config(self, config: Dict) -> Dict:
+
+        child_config = copy.deepcopy(config)
+
+        # pick random hyperparameter and mutate it
+        hypers = []
+        for k, v in self.configspace.items():
+            if isinstance(v, Domain):
+                hypers.append(k)
+        name = self.random_state.choice(hypers)
+        hyperparameter = self.configspace[name]
+
+        if isinstance(hyperparameter, Categorical):
+            # drop current values from potential choices to not sample the same value again
+            choices = [cat for cat in hyperparameter.categories if cat != config[name]]
+            new_value = self.random_state.choice(choices)
+
+        elif isinstance(hyperparameter, Float) or isinstance(hyperparameter, Integer):
+            new_value = hyperparameter.sample(random_state=self.random_state)
+
+        else:
+            raise AssertionError(f"Hyperparameter {name} must be of type Float, Categorical or Integer")
+
+        child_config[name] = new_value
+
+        return child_config
+
+    def sample_random_config(self) -> Dict:
+        return {
+          k: v.sample(random_state=self.random_state) if isinstance(v, Domain) else v
+          for k, v in self.configspace.items()
+        }
+
+    def get_config(self, **kwargs):
+
+        initial_config = self._next_initial_config()
+
+        if initial_config is not None:
+            return initial_config
+
+        if len(self.population) < self.population_size:
+            config = self.sample_random_config()
+        else:
+            candidates = self.random_state.choice(list(self.population), size=self.sample_size)
+            parent = min(candidates, key=lambda i: i.score)
+
+            config = self.mutate_arch(parent.config)
```

Comment:
This one is undefined now (It would make sense to diminish the population size to have this branch this tested)